File: app/src/main/java/com/armageddon/android/flickrdroid/ui/activities/MainActivity.java
package com.armageddon.android.flickrdroid.ui.activities;

import android.app.Activity;
import android.os.Bundle;

import com.armageddon.android.flickrdroid.common.QueryPreferences;


/**
 * Launcher (Main) Activity.
 * Only for the first time launch logo animation (LogoActivity.class) will be started.
 */

public class MainActivity extends Activity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        if (!QueryPreferences.isIntroShown(this)) {
            QueryPreferences.setIntroShown(this, true);
            startActivity(LogoActivity.newIntent(this, true));
        } else {
            startActivity(SearchActivity.newIntent(this,
                    null,
                    SearchActivity.SHOW_MODE,
                    false));
        }
        finish();
    }

}

File: app/src/main/java/com/armageddon/android/flickrdroid/ui/activities/MainActivity.kt
package com.armageddon.android.flickrdroid.ui.activities

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import com.armageddon.android.flickrdroid.common.AppPreferences

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        AppPreferences.setCategoryActivePosition(this,-1)

        /** Show intro for first launch */
        val intent = when (AppPreferences.getUserIconServer(this).isNullOrBlank()) {
            true -> {
                when (AppPreferences.isIntroShown(this)) {
                    true -> {
                        TopPhotoActivity.newIntent(this)
                    }
                    false -> {
                        AppPreferences.setIntroShown(this, true)
                        IntroActivity.newIntent(this, true)
                    }
                }
            }
            false -> {
                FollowingsActivity.newIntent(this, false)
            }
        }.apply {
            addFlags(Intent.FLAG_ACTIVITY_NO_ANIMATION)
        }
        startActivity(intent)
        finish()

    }
}